### DISCUSSION TITLE
access the browser localStorage from leptos

### DISCUSSION THREAD

**Discussion by qknight** [leptos-rs/leptos#1903] [Q&A] [answered]
Is there a way to access the browser localStorage from leptos?

https://blog.logrocket.com/localstorage-javascript-complete-guide/

**Comment by gbj** [comment, marked as answer]
Here are some good options:
1. [`use_local_storage`](https://leptos-use.rs/storage/use_local_storage.html) from the `leptos-use` library
2. [`gloo_storage`](https://docs.rs/gloo-storage/latest/gloo_storage/) from `gloo`, which offers a variety of Rust-friendly wrappers for web APIs
3. Using the `web-sys` Local Storage directly as in the `todomvc` example in this repo https://github.com/leptos-rs/leptos/blob/11f6a5d34126c69d48adbb88e21a9ecbaea22eec/examples/todomvc/src/lib.rs#L12-L23

**Reply by qknight** [reply]
Thanks!!!!

**Reply by dominikwilkowski** [reply]
What about https://docs.rs/leptos/latest/leptos/prelude/struct.LocalStorage.html? It's in the prelude so I would assume this is now included in Leptos?

**Reply by gbj** [reply]
@dominikwilkowski That is not related to the browser's concept of `LocalStorage`; rather it refers to one method for how arena-allocated items like signals can be stored, namely in thread-local storage. The answer given above and the three options included remain the correct answer.

**Reply by dominikwilkowski** [reply]
I got you. Thank you. I went with [gloo_storage](https://crates.io/crates/gloo-storage). I was confused there for a sec with the naming.

**Reply by rafiyq** [reply]
how to specify to only read localstorage in the client side not in the ssr?

**Reply by dominikwilkowski** [reply]
I found this bit helpful: https://book.leptos.dev/ssr/24_hydration_bugs.html?highlight=localSto#not-all-client-code-can-run-on-the-server
